test/modules/generic-mqtt-tester/src/message_channel.rs

```diff
@@ -103,6 +119,12 @@ impl MessageHandler for RelayingMessageHandler {
             "relaying publication with sequence number {}",
             sequence_number,
         );
+
+        // Wait 1 second before relaying to mimic real use cases. This can help
+        // to surface some issues, as connection could drop between original
+        // message receipt and subsequent relay.
+        time::delay_for(self.message_frequency).await;
```

Comment:
@nlcamp Thought of this other improvement we can make.

Reply:
Good idea! 

nit: Suggest calling the parameter "message_interval" or "message_period" for clarity, instead of message_frequency.